### PR TITLE
Migrate snowflake driver repository build to gradle

### PR DESCRIPTION
**Pull request by daneshk**
# Description

Migrate the snowflake driver repository to gradle build.
The main reason for migration is to publish snowflake build artifacts to the GitHub repository. then we can use the artifact to build the snowflake connector with the latest driver connector version without releasing the driver.

Fixes #ballerina-platform/ballerina-library/issues/5905

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -42,3 +40,6 @@ hs_err_pid*
 
 # Ignore Gradle build output directory
 build
+
+# Ignore `docker.env` file which passes the env from the host machine to the docker container
+*.env
```
Comment (daneshk) [inline .gitignore:45]:
```suggestion
*.env

```

File: settings.gradle [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+include ':snowflake.driver-ballerina'
+
+project(':snowflake.driver-ballerina').projectDir = file("ballerina")
+
+gradleEnterprise {
+    buildScan {
+        termsOfServiceUrl = 'https://gradle.com/terms-of-service'
+        termsOfServiceAgree = 'yes'
+    }
+}
```
Comment (daneshk) [inline settings.gradle:23]:
```suggestion
}

```

File: .github/workflows/build-timestamped-master.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: Build
```
Comment (niveathika) [inline .github/workflows/build-timestamped-master.yml:1]: Cant we reuse the connector templates for the workflows? @ThisaruGuruge

Comment (daneshk) [reply .github/workflows/build-timestamped-master.yml:1]: I just copied them from the `h2.driver`. I will check

File: README.md [hunk 4]
```diff
@@ -9,46 +9,56 @@ Ballerina Snowflake Driver Library
...
 
 ## Code of conduct
 
-All the contributors are encouraged to read the [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
+All contributors are encouraged to read the [Ballerina code of conduct](https://ballerina.io/code-of-conduct).
 
 ## Useful links
 
-* Discuss the code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
+* For more information go to the [`h2.driver` library](https://lib.ballerina.io/ballerinax/h2.driver/latest).
```
Comment (Bhashinee) [inline README.md:62]:
```suggestion
* For more information go to the [`snowflake.driver` library](https://lib.ballerina.io/ballerinax/snowflake.driver/latest).
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -1,47 +1,17 @@
...
 
-# Controls when the action will run.
-# Triggers the workflow on push.
 on:
   push:
     branches:
       - main
-      - 2201.[0-9]+.x
+  repository_dispatch:
+    types: check_connector_for_breaking_changes
```
Comment (niveathika) [inline .github/workflows/ci.yml:8]:
```suggestion
```
No need to check for drivers as there will be no breaking changes

**Review by niveathika — DISMISSED** [review]
Other than the minor comment. LGTM

File: .github/ISSUE_TEMPLATE/type_bug.md
Comment (ThisaruGuruge) [inline .github/ISSUE_TEMPLATE/type_bug.md:1]: Do we need the issue templates in these repositories?

File: ballerina/build.gradle [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            credentials {
+                username System.getenv('packageUser')
+                password System.getenv('packagePAT')
+            }
+        }
+    }
+    dependencies {
+        classpath "io.ballerina:plugin-gradle:${project.ballerinaGradlePluginVersion}"
+    }
+}
```
Comment (ThisaruGuruge) [inline ballerina/build.gradle:35]: Shall we move the dependency management to the `settings.gradle` file? Check other repos like [gmail](https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/blob/e211d4e622c4b4a43f5d47a4b0e06e74297ca4b1/settings.gradle#L14).

File: ballerina/build.gradle [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        return extVersion.replace("${project.ext.snapshotVersion}", '')
+    }
+}
+
+apply plugin: 'io.ballerina.plugin'
+
+ballerina {
+    packageOrganization = packageOrg
+    module = packageName
+    langVersion = ballerinaLangVersion
```
Comment (ThisaruGuruge) [inline ballerina/build.gradle:65]: We need to provide `isConnector` parameter here.

File: .github/ISSUE_TEMPLATE/type_bug.md
Comment (ThisaruGuruge) [inline .github/ISSUE_TEMPLATE/type_bug.md:1]: Do we need the issue templates in these repositories?

[the remark this answers is not in the document]
Comment (daneshk) [reply .github/ISSUE_TEMPLATE/type_bug.md:1]: Yes, removed in [1c26c36](https://github.com/ballerina-platform/module-ballerinax-snowflake.driver/pull/43/commits/1c26c369337d5db1169fd0c815698db8d21fbc83)

File: ballerina/build.gradle (hunk 7 above)
Comment (daneshk) [reply ballerina/build.gradle:65]: for the driver, is this true or false?

File: ballerina/build.gradle [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    }
+}
+
+apply plugin: 'io.ballerina.plugin'
+
+ballerina {
+    packageOrganization = packageOrg
+    module = packageName
+    isConnector = true
+    platform = "any"
```
Comment (ThisaruGuruge) [inline ballerina/build.gradle:55]: Since this has a Jar dependency, it should be `java17`, right?

```suggestion
    platform = "java17"
```

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Pull request merged**